src/hypothesis/strategies.py

```diff
@@ -1374,4 +1366,38 @@ def check_valid_sizes(min_size, average_size, max_size):
 
 
 _AVERAGE_LIST_LENGTH = 5.0
+
+
+def deferred(definition):
+    """A deferred strategy allows you to write a strategy that references other
+    strategies that have not yet been defined. This allows for the easy
+    definition of recursive and mutually recursive strategies.
+
+    The definition argument should be a zero-argument function that returns a
+    strategy. It will be evaluated the first time the strategy is used to
+    produce an example.
+
+    Example usage:
```

Comment:
I would like these to be in a `.. doctest::` block, so they'll be tested once I sort that out.